app/src/main/kotlin/com/wire/android/util/deeplink/DeepLinkProcessor.kt

```diff
@@ -48,47 +52,108 @@ sealed class DeepLinkResult {
     }
 
     data class OpenConversation(
-        val conversationsId: ConversationId,
+        val conversationId: ConversationId,
         val switchedAccount: Boolean = false
     ) : DeepLinkResult()
 
-    data class OpenOtherUserProfile(val userId: QualifiedID, val switchedAccount: Boolean = false) :
-        DeepLinkResult()
+    data class OpenOtherUserProfile(val userId: QualifiedID, val switchedAccount: Boolean = false) : DeepLinkResult()
 
-    data class JoinConversation(val code: String, val key: String, val domain: String?) :
-        DeepLinkResult()
+    data class JoinConversation(
+        val code: String,
+        val key: String,
+        val domain: String?,
+        val switchedAccount: Boolean = false
+    ) : DeepLinkResult()
 
     data class MigrationLogin(val userHandle: String) : DeepLinkResult()
+
+    data object SharingIntent : DeepLinkResult()
+
+    data object AuthorizationNeeded : DeepLinkResult()
+
+    data class Failure(val reason: FailureReason) : DeepLinkResult()
+
+    enum class FailureReason {
+        OngoingCall,
+        Unknown
+    }
```

Comment:
```suggestion
    sealed class Failure : DeepLinkResult() {
        data object OngoingCall: Failure
        data object Unknown: Failure
    }
```

IMO `DeepLinkResult.Failure.OngoingCall` looks better then `DeepLinkResult.Failure(DeepLinkResult.FailureReason.OngoingCall)`

Reply:
> IMO `DeepLinkResult.Failure.OngoingCall` looks better then `DeepLinkResult.Failure(DeepLinkResult.FailureReason.OngoingCall)`

Agreed.

Just one addition suggestion: instead of `sealed class`, it can be a `sealed interface`. Less parenthesis and can be a bit more flexible.